PowerEditor/src/WinControls/AboutDlg/AboutDlg.cpp

```diff
@@ -381,10 +218,191 @@ void DebugInfoDlg::doDialog()
+		szProductName[sizeof(szProductName) / sizeof(TCHAR) - 1] = '\0';
+
+		dataSize = sizeof(szReleaseId);
+		RegQueryValueExW(hKey, TEXT("ReleaseId"), NULL, NULL, reinterpret_cast<LPBYTE>(szReleaseId), &dataSize);
+		szReleaseId[sizeof(szReleaseId) / sizeof(TCHAR) - 1] = '\0';
+
+		dataSize = sizeof(szCurrentBuildNumber);
+		RegQueryValueExW(hKey, TEXT("CurrentBuildNumber"), NULL, NULL, reinterpret_cast<LPBYTE>(szCurrentBuildNumber), &dataSize);
+		szCurrentBuildNumber[sizeof(szCurrentBuildNumber) / sizeof(TCHAR) - 1] = '\0';
+
+		dataSize = sizeof(DWORD);
+		if (RegQueryValueExW(hKey, TEXT("UBR"), NULL, NULL, reinterpret_cast<LPBYTE>(&dwUBR), &dataSize) == ERROR_SUCCESS)
+		{
+			generic_sprintf(szUBR, TEXT("%u"), dwUBR);
+		}
+
+		RegCloseKey(hKey);
+	}
+
+	// Get alternative OS information
+	if (szProductName[0] == '\0')
+	{
+		generic_sprintf(szProductName, TEXT("%s"), (NppParameters::getInstance()).getWinVersionStr().c_str());
+	}
+
+	// Override ProductName if it's Windows 11
+	if (NppDarkMode::isWindows11())
+		generic_sprintf(szProductName, TEXT("%s"), TEXT("Windows 11"));
+
+	if (szCurrentBuildNumber[0] == '\0')
+	{
+		DWORD dwVersion = GetVersion();
+		if (dwVersion < 0x80000000)
+		{
+			generic_sprintf(szCurrentBuildNumber, TEXT("%u"), HIWORD(dwVersion));
+		}
+	}
+
+	_debugInfoStr += TEXT("OS Name : ");
+	_debugInfoStr += szProductName;
+	_debugInfoStr += TEXT(" (");
+	_debugInfoStr += (NppParameters::getInstance()).getWinVerBitStr();
+	_debugInfoStr += TEXT(") ");
+	_debugInfoStr += TEXT("\r\n");
+
+	if (szReleaseId[0] != '\0')
+	{
+		_debugInfoStr += TEXT("OS Version : ");
+		_debugInfoStr += szReleaseId;
+		_debugInfoStr += TEXT("\r\n");
+	}
+
+	if (szCurrentBuildNumber[0] != '\0')
+	{
+		_debugInfoStr += TEXT("OS Build : ");
+		_debugInfoStr += szCurrentBuildNumber;
+		_debugInfoStr += TEXT(".");
+		_debugInfoStr += szUBR;
+		_debugInfoStr += TEXT("\r\n");
+	}
+
+	{
+		TCHAR szACP[32];
+		generic_sprintf(szACP, TEXT("%u"), ::GetACP());
+		_debugInfoStr += TEXT("Current ANSI codepage : ");
+		_debugInfoStr += szACP;
+		_debugInfoStr += TEXT("\r\n");
+	}
+
+	// Detect WINE
+	PWINEGETVERSION pWGV = nullptr;
+	HMODULE hNtdllModule = GetModuleHandle(L"ntdll.dll");
+	if (hNtdllModule)
+	{
+		pWGV = (PWINEGETVERSION)GetProcAddress(hNtdllModule, "wine_get_version");
+	}
+
+	if (pWGV != nullptr)
+	{
+		TCHAR szWINEVersion[32];
+		generic_sprintf(szWINEVersion, TEXT("%hs"), pWGV());
+
+		_debugInfoStr += TEXT("WINE : ");
+		_debugInfoStr += szWINEVersion;
+		_debugInfoStr += TEXT("\r\n");
+	}
+
+	// Plugins
+	_debugInfoStr += TEXT("Plugins : ");
+	_debugInfoStr += _loadedPlugins.length() == 0 ? TEXT("none") : _loadedPlugins;
+	_debugInfoStr += TEXT("\r\n");
+
+	// Set Debug Info text and leave the text in selected state
+	::SetDlgItemText(_hSelf, IDC_DEBUGINFO_EDIT, _debugInfoStr.c_str());
+	::SendDlgItemMessage(_hSelf, IDC_DEBUGINFO_EDIT, EM_SETSEL, 0, _debugInfoStr.length() - 1);

```

Comment:
Line 403 selects the entire Debug Info text string after refreshing it. This selection action preserves the exiting behavior in NPP.